### PR TITLE
Refactor capitalizeEachWord and re-enable logging

### PR DESCRIPTION
**Pull request by ndsprinkle**
Refactor capitalizeEachWord to not return an unrelated string when a non-string is supplied and also re-enable the logging to Sentry for this non-supplied string error.

## Description
capitalizeEachWord was logging a message to Sentry each time a non-string was passed into the function. This spawned https://github.com/department-of-veterans-affairs/va.gov-team/issues/1365. This message was logging frequently throughout the 526 form if a user entered a blank condition because capitalizeEachWord was called each time the condition was shown. At some point, this logging message was commented out, so this story (1365) pivoted to being about re-enabling this logging message and reworking capitalizeEachWord and the places it is used to no longer cause the message to be logged as frequently. The function was also reworked to no longer return an unrelated string, but rather return null if a non-string is provided. 

## Testing done
Test were modified and re-run.

## Acceptance criteria
- [ ] Re-enable Sentry logging for capitalizeEachWord
- [ ] Refactor capitalizeEachWord to not return an unrelated string when a non-string is provided

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ const formConfig = {
           schema: { type: 'object', properties: {} },
         },
         newDisabilityFollowUp: {
-          title: formData => capitalizeEachWord(formData.condition),
+          title: formData => typeof formData.condition === 'string' ? capitalizeEachWord(formData.condition) : 'Unknown Condition',
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/config/form.js]: When is `form.condition` not a string? could this check just be 

```
form.condition ? capitalizeEachWord(formData.condition) : 'Unknown Condition'
```

**Review by rianfowler — APPROVED** [review]
We have some code review guidelines [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md). This should have an approval from a teammate before being moved out of draft PR state. 

Only had one question. I haven't worked on this form so I'm not familiar the purpose of some of these calls.

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 2]
```diff
@@ -165,14 +165,12 @@ export const capitalizeEachWord = name => {
   }
 
   if (typeof name !== 'string') {
-    // Sentry.captureMessage(
-    //   `form_526_v1 / form_526_v2: capitalizeEachWord requires 'name' argument of type 'string' but got ${typeof name}`,
-    // );
+    Sentry.captureMessage(
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/utils.jsx]: :man_shrugging: My two cents is to nix this whole chunk and handle the returned `null` appropriately from where this function is called without doing a type check each time.
```js
const capitalizedDisabilityName = capitalizeEachWord(disabilityName) || UNKNOWN_CONDITION;
```

If I recall, `Sentry.captureMessage()` doesn't return a stacktrace, so this wouldn't be useful for hunting down _where_ we're passing `capitalizeEachWord` a non-string. (I could be wrong there, mind you, but I _think_ all we get is the breadcrumbs from the browser—what buttons they poked, etc.)

File: src/applications/disability-benefits/all-claims/pages/newDisabilityFollowUp.js [hunk 3]
```diff
@@ -31,10 +31,14 @@ const getDisabilitiesList = createSelector(
...
       .filter((item, index) => index !== currentIndex)
-      .map(item => capitalizeEachWord(item.condition));
+      .map((item) => {
+        return typeof item.condition === 'string' ? capitalizeEachWord(item.condition) : 'Unknown Condition';
+      });
 
     return ratedDisabilities
-      .map(disability => capitalizeEachWord(disability.name))
+      .map((disability) => {
+        return typeof disability.name === 'string' ? capitalizeEachWord(disability.name) : 'Unknown Condition'
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/newDisabilityFollowUp.js]: I recommend using a constant for `'Unknown Condition'` for consistency.

**Pull request merged**